diff --git a/include/kimera-vio/mesh/MeshOptimization.h b/include/kimera-vio/mesh/MeshOptimization.h
index fa1da651..7d73d93e 100644
--- a/include/kimera-vio/mesh/MeshOptimization.h
+++ b/include/kimera-vio/mesh/MeshOptimization.h
@@ -21,6 +21,7 @@
 #include <Eigen/Core>
 
 #include <opencv2/opencv.hpp>
+#include <opencv2/viz.hpp>
 
 #include <gtsam/geometry/Cal3_S2.h>
 #include <gtsam/geometry/Pose3.h>
diff --git a/include/kimera-vio/utils/UtilsNumerical.h b/include/kimera-vio/utils/UtilsNumerical.h
index 47fb8c9e..2c21f977 100644
--- a/include/kimera-vio/utils/UtilsNumerical.h
+++ b/include/kimera-vio/utils/UtilsNumerical.h
@@ -16,6 +16,7 @@
 
 #include <algorithm>
 #include <iostream>
+#include <vector>
 
 namespace VIO {
 
diff --git a/src/frontend/RgbdCamera.cpp b/src/frontend/RgbdCamera.cpp
index 761567e7..29344a7f 100644
--- a/src/frontend/RgbdCamera.cpp
+++ b/src/frontend/RgbdCamera.cpp
@@ -15,6 +15,8 @@
 
 #include "kimera-vio/frontend/RgbdCamera.h"
 
+#include <opencv2/viz.hpp>
+
 #include "kimera-vio/frontend/Camera.h"
 
 namespace VIO {
