diff --git a/python/build_targets.bzl b/python/build_targets.bzl
index fff32b4a5..08e0212cf 100644
--- a/python/build_targets.bzl
+++ b/python/build_targets.bzl
@@ -84,7 +84,7 @@ def build_targets(name):
         ],
         deps = select({
             "//conditions:default": [],
-            ":use_fast_cpp_protos": ["//external:python_headers"],
+            ":use_fast_cpp_protos": ["@rules_python//python/cc:current_py_cc_headers"],
         }),
     )
 
@@ -123,7 +123,7 @@ def build_targets(name):
             "//src/google/protobuf:descriptor_legacy",
         ] + select({
             "//conditions:default": [],
-            ":use_fast_cpp_protos": ["//external:python_headers"],
+            ":use_fast_cpp_protos": ["@rules_python//python/cc:current_py_cc_headers"],
         }),
     )
 
@@ -386,7 +386,7 @@ def build_targets(name):
         hdrs = ["google/protobuf/proto_api.h"],
         visibility = ["//visibility:public"],
         deps = [
-            "//external:python_headers",
+            "@rules_python//python/cc:current_py_cc_headers",
         ],
     )
 
