diff --git a/BUILD.bazel b/BUILD.bazel
index 23efee4a3..6908bf120 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,5 +1,7 @@
 # Bazel (https://bazel.build/) BUILD file for Protobuf.
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_proto_library")
 load("@rules_java//java:defs.bzl", "java_lite_proto_library", "java_proto_library")
 load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
diff --git a/python/BUILD.bazel b/python/BUILD.bazel
index ea1dac316..b4c169891 100644
--- a/python/BUILD.bazel
+++ b/python/BUILD.bazel
@@ -23,6 +23,7 @@ package(
 #     default_applicable_licenses = ["//upb:license"],
     # end:google_only
     default_visibility = ["//python/dist:__pkg__"],
+    features=["-warning_compile_flags"]
 )
 
 LIMITED_API_FLAG_SELECT = {
diff --git a/src/google/protobuf/BUILD.bazel b/src/google/protobuf/BUILD.bazel
index 8961ca6af..5dbd09c3b 100644
--- a/src/google/protobuf/BUILD.bazel
+++ b/src/google/protobuf/BUILD.bazel
@@ -13,6 +13,7 @@ package(
         "//:__pkg__",  # "public" targets are alias rules in //.
         "//json:__subpackages__",
     ],
+    features=["-warning_compile_flags"]
 )
 
 proto_library(
diff --git a/src/google/protobuf/compiler/BUILD.bazel b/src/google/protobuf/compiler/BUILD.bazel
index 273dae00b..9742520f7 100644
--- a/src/google/protobuf/compiler/BUILD.bazel
+++ b/src/google/protobuf/compiler/BUILD.bazel
@@ -2,6 +2,8 @@
 # Protocol Buffers Compiler
 ################################################################################
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_proto_library", "cc_test")
 load(
     "@rules_pkg//:mappings.bzl",
diff --git a/src/google/protobuf/compiler/cpp/BUILD.bazel b/src/google/protobuf/compiler/cpp/BUILD.bazel
index 8ca37b4b6..9b2ed350d 100644
--- a/src/google/protobuf/compiler/cpp/BUILD.bazel
+++ b/src/google/protobuf/compiler/cpp/BUILD.bazel
@@ -2,6 +2,8 @@
 # Protocol Buffers Compiler - C++ code generator
 ################################################################################
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_library", "cc_proto_library", "cc_test")
 load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
 load("@rules_proto//proto:defs.bzl", "proto_library")
diff --git a/src/google/protobuf/compiler/csharp/BUILD.bazel b/src/google/protobuf/compiler/csharp/BUILD.bazel
index 414dd0ffd..474a71755 100644
--- a/src/google/protobuf/compiler/csharp/BUILD.bazel
+++ b/src/google/protobuf/compiler/csharp/BUILD.bazel
@@ -2,6 +2,8 @@
 # Protocol Buffers Compiler - C# code generator
 ################################################################################
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
 load("//build_defs:cpp_opts.bzl", "COPTS")
diff --git a/src/google/protobuf/compiler/java/BUILD.bazel b/src/google/protobuf/compiler/java/BUILD.bazel
index a6e78f009..15a92c87c 100644
--- a/src/google/protobuf/compiler/java/BUILD.bazel
+++ b/src/google/protobuf/compiler/java/BUILD.bazel
@@ -2,6 +2,8 @@
 # Protocol Buffers Compiler - Java code generator
 ################################################################################
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
 load("//build_defs:cpp_opts.bzl", "COPTS")
diff --git a/src/google/protobuf/compiler/objectivec/BUILD.bazel b/src/google/protobuf/compiler/objectivec/BUILD.bazel
index 5ab3381af..565b9b232 100644
--- a/src/google/protobuf/compiler/objectivec/BUILD.bazel
+++ b/src/google/protobuf/compiler/objectivec/BUILD.bazel
@@ -2,6 +2,8 @@
 # Protocol Buffers Compiler - ObjC code generator
 ################################################################################
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
 load("//build_defs:cpp_opts.bzl", "COPTS")
diff --git a/src/google/protobuf/compiler/php/BUILD.bazel b/src/google/protobuf/compiler/php/BUILD.bazel
index f814a03a7..c175e2d58 100644
--- a/src/google/protobuf/compiler/php/BUILD.bazel
+++ b/src/google/protobuf/compiler/php/BUILD.bazel
@@ -2,6 +2,8 @@
 # Protocol Buffers Compiler - PHP code generator
 ################################################################################
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_library")
 load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
 load("//build_defs:cpp_opts.bzl", "COPTS")
diff --git a/src/google/protobuf/compiler/ruby/BUILD.bazel b/src/google/protobuf/compiler/ruby/BUILD.bazel
index d01764c2f..0e1429bb7 100644
--- a/src/google/protobuf/compiler/ruby/BUILD.bazel
+++ b/src/google/protobuf/compiler/ruby/BUILD.bazel
@@ -2,6 +2,8 @@
 # Protocol Buffers Compiler - Ruby code generator
 ################################################################################
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
 load("//build_defs:cpp_opts.bzl", "COPTS")
diff --git a/src/google/protobuf/compiler/rust/BUILD.bazel b/src/google/protobuf/compiler/rust/BUILD.bazel
index f404beae9..7df1b57d1 100644
--- a/src/google/protobuf/compiler/rust/BUILD.bazel
+++ b/src/google/protobuf/compiler/rust/BUILD.bazel
@@ -2,6 +2,8 @@
 # Protocol Buffers Compiler - Rust code generator
 ################################################################################
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_library")
 load("//build_defs:cpp_opts.bzl", "COPTS")
 
diff --git a/src/google/protobuf/json/BUILD.bazel b/src/google/protobuf/json/BUILD.bazel
index f45c4fab2..354f2ec88 100644
--- a/src/google/protobuf/json/BUILD.bazel
+++ b/src/google/protobuf/json/BUILD.bazel
@@ -1,10 +1,13 @@
 load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("//build_defs:cpp_opts.bzl", "COPTS")
 
-package(default_visibility = [
-    "//pkg:__pkg__",
-    "//src/google/protobuf/json:__pkg__",
-])
+package(
+    default_visibility = [
+        "//pkg:__pkg__",
+        "//src/google/protobuf/json:__pkg__",
+    ],
+    features=["-warning_compile_flags"],
+)
 
 licenses(["notice"])
 
diff --git a/src/google/protobuf/compiler/python/BUILD.bazel b/src/google/protobuf/compiler/python/BUILD.bazel
index 64b5808cb..2b629e841 100644
--- a/src/google/protobuf/compiler/python/BUILD.bazel
+++ b/src/google/protobuf/compiler/python/BUILD.bazel
@@ -2,6 +2,8 @@
 # Protocol Buffers Compiler - Python code generator
 ################################################################################
 
+package(features=["-warning_compile_flags"])
+
 load("@rules_cc//cc:defs.bzl", "cc_library", "cc_test")
 load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
 load("//build_defs:cpp_opts.bzl", "COPTS")
